### PR TITLE
Add cron recipe to allow per-app scheduled tasks.

### PR DESCRIPTION
**Pull request by berkes**
Note: WIP.

This adds a recipe that schedules tasks for apps. It simply adds entries to the root crontab
with commands that will be ran within the context of the app.

Mostly a simple wrapper around the cron resource; so we can define per-app scheduled tasks.

* They will be ran by `deploy` user.
* The command will be changed to become `cd {path_to_app}/current && ( PATH=/opt/rbenv/shims:$PATH RAILS_ENV={your_env} bundle exec {your_command} )`
* All other attributes from [http://docs.chef.io/resource_cron.html](http://docs.chef.io/resource_cron.html) will be passed along to the `cron` resource.

### CODE REVIEW

**Comment by jvanbaarsen** [comment]
@berkes Maybe interesting to look at https://github.com/intercity/chef-repo/blob/master/vendor/cookbooks/backups/recipes/default.rb#L83 since I already implemented a cron cookbook for the backup feature.

**Comment by jvanbaarsen** [comment]
For your concerns:

**Is this feature needed?** 
I believe this would make a great addition!

**Do we need a cleaner or more predictable outcome for the command?** 
Can you explain in a bit more detail what you mean with cleaner?

**Other comments?**
Awesome work :D :cake:

**Comment by berkes** [comment]
> since I already implemented a cron cookbook for the backup feature.

I was aware of that and my recipe uses the same underlying `cron` LWRP. 

The two cron-tasks are a bit different, though. 
These new custom tasks are "just any command" and ran by the deploy user (i.e. they are added to the crontab for the deploy user)
Your crontab runs as root and is a fixed command. 

> Can you explain in a bit more detail what you mean with cleaner?

When you add `"command": "rake app:cleanup_old_orders"` under "intercity_example" app, my cron recipe will add an entry to the deploy user with the command `+ command = "cd /u/apps/intercity_example && ( PATH=/opt/rbenv/shims:$PATH RAILS_ENV=production bundle exec rake app:cleanup )"

In other words: it adds a lot of extra commands and cruft to the command a user entered. This might cause weird behaviour when the user adds complex commands himself.

For now, I'd say a warning in documentation is in place, but we might want to add a user-settable flag to avoid adding our own commands around the users' command in future.

**Comment by berkes** [comment]
@jvanbaarsen @michiels how do you think this should best be integrated?

How to include it:
* as a recipe under the 'rails' cookbook?
* as a separate cookbook (as proposed in this PR, but moving to the 'rails' cookbook is easy)

Where or when to run it:
* as a role?
* as an includable recipe?
* default included in the rails::setup recipe?
* default included in the base role? (ugh, ugly)
* default included in both the rails and rails_passenger roles?

IMO this should simply always be ran, therefore I don't really like the options where people have to include these as roles or recipes themselves.
